Framework/src/InfrastructureGenerator.cxx

```diff
@@ -449,11 +452,25 @@ void InfrastructureGenerator::generateCheckRunners(framework::WorkflowSpec& work
     ILOG(Info, Devel) << ENDM;
 
     if (!checks.empty()) { // Create a CheckRunner for the grouped checks
-      workflow.emplace_back(checkRunnerFactory.create(checks, configurationSource, storeVectorMap[inputNames]));
+      DataProcessorSpec spec = checkRunnerFactory.create(checks, configurationSource);
+      workflow.emplace_back(spec);
+      checkRunnerOutputs.insert(checkRunnerOutputs.end(), spec.outputs.begin(), spec.outputs.end());
     } else { // If there are no checks, create a sink CheckRunner
-      workflow.emplace_back(checkRunnerFactory.createSinkDevice(tasksOutputMap.find(inputNames[0])->second, configurationSource));
+      DataProcessorSpec spec = checkRunnerFactory.createSinkDevice(tasksOutputMap.find(inputNames[0])->second, configurationSource);
+      workflow.emplace_back(spec);
+      checkRunnerOutputs.insert(checkRunnerOutputs.end(), spec.outputs.begin(), spec.outputs.end());
     }
   }
+
+  ILOG(Info) << ">> Outputs (" << checkRunnerOutputs.size() << "): ";
+  for (const auto& output : checkRunnerOutputs)
+    ILOG(Info) << DataSpecUtils::describe(output) << " ";
+  ILOG(Info) << ENDM;
+
+  // as we need information about the checkRunners we do it here instead of a separate static method
+  AggregatorRunnerFactory aggregatorRunnerFactory;
+  DataProcessorSpec spec = aggregatorRunnerFactory.create(checkRunnerOutputs, configurationSource);
+  workflow.emplace_back(spec);
```

Comment:
I agree. I have extracted the static method. However, to avoid recomputing twice the same thing I am returning from generateCheckRunners the list of outputs and passing it to the generateAggregators. I have put a note to look into it again.